### PR TITLE
Feature/add library to cli

### PR DESCRIPTION
**Pull request by outsidecoder**
### What
Added Library functionality to dp tooling under project-generation.

### How to review
Build and run and select 5 for library and check it creates the correct files for library.

### Who can review
Anyone with Go knowledge.

### CODE REVIEW

File: project_generation/content/templates/base-app/Makefile.tmpl [hunk 1]
```diff
@@ -11,7 +11,7 @@ all: audit test build
 
 .PHONY: audit
 audit:
-	nancy go.sum
+	go list -m all | nancy sleuth
```
Comment (Marianasilvago) [inline project_generation/content/templates/base-app/Makefile.tmpl:14]: Should we be doing `go list -json -m all` here. Ref: https://github.com/sonatype-nexus-community/nancy.

Also should we consider introducing the facility to have few exclusion of vulnerabilities?
for eg:
`go list -json -m all | nancy sleuth --exclude-vulnerability-file ./.nancy-ignore`

File: project_generation/content/templates/event/Makefile.tmpl [hunk 2]
```diff
@@ -11,7 +11,7 @@ all: audit test build
 
 .PHONY: audit
 audit:
-	nancy go.sum
+	go list -m all | nancy sleuth
```
Comment (Marianasilvago) [inline project_generation/content/templates/event/Makefile.tmpl:14]: same as above

**Review by Marianasilvago — CHANGES_REQUESTED** [review]

File: project_generation/content/templates/base-app/Makefile.tmpl (hunk 1 above)
Comment (outsidecoder) [reply project_generation/content/templates/base-app/Makefile.tmpl]: makes sense thanks

File: project_generation/content/templates/event/Makefile.tmpl (hunk 2 above)
Comment (outsidecoder) [reply project_generation/content/templates/event/Makefile.tmpl:14]: makes sense thanks

**Review by Marianasilvago — APPROVED** [review]

**Pull request merged**